src/Microsoft.ML.OnnxTransform/DnnImageFeaturizerTransform.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.StaticPipe;
+using Microsoft.ML.StaticPipe.Runtime;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// This is a helper class that is required to use the <see cref="DnnImageFeaturizerEstimator"/>.
+    /// Note that by default, it is not usable as it does not have any valid methods that return an <see cref="EstimatorChain{TLastTransformer}"/>
+    /// that is used by the DnnImageFeaturizeEstimator.
+    /// In order to use this, at least one model project with the corresponding extension methods must by included.
+    /// See Microsoft.ML.DNNImageFeaturizer.ResNet18 for an example.
+    /// </summary>
+    public sealed class DnnImageModelSelector
+    {
+    }
+
+    /// <summary>
+    /// The Dnn Image Featurizer is just a wrapper around two <see cref="OnnxScoringEstimator"/>s with present pretrained DNN models.
+    /// Note that because of this, it only works on Windows machines as that is a constraint of the OnnxTransform.
+    /// </summary>
+    public sealed class DnnImageFeaturizerEstimator : IEstimator<TransformerChain<OnnxTransform>>
```

Comment:
What about this is specific to:

1. Deep neural nets, vs any kind of NN?
2. Images?

Couldn't this type have a more general name, so I could use it with audio, and a non-deep NN?

Reply:
Nothing in particular, although currently that is the only use for this transform based on the currently available models which makes the name more descriptive for users trying to figure out what it's for. But with more extensions in the future it could definitely be used for more things. Not sure how likely that is to happen however, and whether it is worth it to give false expectations.

---
In reply to: [232779827](https://github.com/dotnet/machinelearning/pull/1447#discussion_r232779827) [](ancestors = 232779827)